src/main/java/iggly/duke/Storage.java

```diff
@@ -0,0 +1,87 @@
+package iggly.duke;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+
+import iggly.model.Task;
+
+/**
+ * The {@link Storage} class is responsible for handling the reading, writing and saving task data to a file.
+ */
+public class Storage {
+    private final File file;
+
+    /**
+     * Constructs a {@link Storage} instance with the specified file path.
+     *
+     * @param filePath The file path where task data is stored or will be stored.
+     */
+    public Storage(String filePath) {
+        this.file = new File(filePath);
+    }
+
+    /**
+     * Checks if the file associated with this storage exists.
+     *
+     * @return {@code true} if the file exists, {@code false} otherwise.
+     */
+    public boolean isFileExists() {
+        try (FileInputStream fileInputStream = new FileInputStream(file)) {
+            return true;
+        } catch (FileNotFoundException e) {
+            return false;
+        } catch (IOException e) {
+            System.out.println("Error checking file existence: " + e);
+            return false;
+        }
+    }
+
+    /**
+     * Creates a new file at the specified file path.
+     */
+    public void createNewFile() {
+        try {
+            this.file.createNewFile();
+        } catch (IOException e) {
+            System.out.println("Unable to create new file.");
+        }
+    }
+
+    /**
+     * Updates the file with the provided task list.
+     *
+     * @param taskList The list of tasks to be written to the file.
+     */
+    public void update(ArrayList<Task> taskList) {
```

Comment:
updateList maybe? Again by the textbook: Update what? Could be clearer for comprehension upon first look!

Reply:
okie thanks for the heads up!